collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_pgsched.py

```diff
@@ -0,0 +1,435 @@
+                blackout = {'start': blackout_start, 'end': blackout_end}
+                try:
+                    array.set_pgroup(module.params['name'],
+                                     replicate_enabled=replicate_enabled)
+                    array.set_pgroup(module.params['name'],
+                                     replicate_frequency=replicate_frequency,
+                                     replicate_at=replicate_at,
+                                     replicate_blackout=blackout)
+                    array.set_pgroup(module.params['name'],
+                                     target_days=target_days,
+                                     target_per_day=target_per_day,
+                                     target_all_for=target_all_for)
+                    changed = True
+                except Exception:
+                    module.fail_json(msg='Failed to change replication schedule for pgroup {0}.'.format(module.params['name']))
+
+    module.exit_json(changed=changed)
+
+
+def delete_schedule(module, array):
+    """ Delete, ie. disable, Protection Group Schedules"""
+    changed = True
+    if not module.check_mode:
+        try:
+            if module.params['schedule'] == "replication":
+                array.set_pgroup(module.params['name'], replicate_enabled=False)
+                array.set_pgroup(module.params['name'], target_days=0, target_per_day=0,
+                                 target_all_for=0)
+                array.set_pgroup(module.params['name'], replicate_frequency=3600,
+                                 replicate_blackout={'end': 0, 'start': 0})
+            else:
+                array.set_pgroup(module.params['name'], snapshot_enabled=False)
+                array.set_pgroup(module.params['name'], days=0, per_day=0, all_for=0)
+                array.set_pgroup(module.params['name'], snap_frequency=300)
+        except Exception:
+            module.fail_json(msg='Deleting pgroup {0} {1} schedule failed.'.format(module.params['name'],
+                                                                                   module.params['schedule']))
+
+    module.exit_json(changed=changed)
+
+
+def main():
+    argument_spec = purefa_argument_spec()
+    argument_spec.update(dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['absent', 'present']),
+        schedule=dict(type='str', required=True, choices=['replication', 'snapshot']),
+        blackout_start=dict(type='str'),
+        blackout_end=dict(type='str'),
+        snap_at=dict(type='int'),
+        replicate_at=dict(type='str'),
+        replicate_frequency=dict(type='int'),
+        snap_frequency=dict(type='int'),
+        all_for=dict(type='int'),
+        days=dict(type='int'),
+        per_day=dict(type='int'),
+        target_all_for=dict(type='int'),
+        target_per_day=dict(type='int'),
+        target_days=dict(type='int'),
+        enabled=dict(type='bool', default=True)
```

Comment:
I tried to use the same values as in the examples and they just don't work because the module doesn't accept parameters: `replicate_enabled` and `snap_enabled`. Output error:

```
failed: [example.com] (item={'name': 'test-vg', 'volumes': [{'name': 'test-vol', 'size': '5G'}], 'pg': {'replication': {'enabled': False, 'frequency': 86400, 'at': 55800, 'per_day': 5, 'all_for': 5, 'blackout_start': '2AM', 'blackout_end': '5AM'}, 'snapshot': {'enabled': False, 'frequency': 86400, 'at': 55800, 'per_day': 5, 'all_for': 5}}}) => changed=false 
  ansible_loop_var: item
  invocation:
    module_args:
      api_token: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
      blackout_end: 5AM
      blackout_start: 2AM
      fa_url: example.com
      name: test-vg
      replicate_at: '55800'
      replicate_enabled: false
      replicate_frequency: '86400'
      schedule: replication
      target_all_for: '5'
      target_per_day: '5'
  msg: 'Unsupported parameters for (purestorage.flasharray.purefa_pgsched) module: replicate_enabled Supported parameters include: all_for, api_token, blackout_end, blackout_start, days, enabled, fa_url, name, per_day, replicate_at, replicate_frequency, schedule, snap_at, snap_frequency, state, target_all_for, target_days, target_per_day'
```

Reply:
But even after changing the parameters to `enabled` it threw an error:

```
failed: [example.com] (item={'name': 'test-vg', 'volumes': [{'name': 'test-vol', 'size': '5G'}], 'pg': {'replication': {'enabled': False, 'frequency': 86400, 'at': 55800, 'per_day': 5, 'all_for': 5, 'blackout_start': '2AM', 'blackout_end': '5AM'}, 'snapshot': {'enabled': False, 'frequency': 86400, 'at': 55800, 'per_day': 5, 'all_for': 5}}}) => changed=false 
  ansible_loop_var: item
  item:
    name: test-vg
    pg:
      replication:
        all_for: 5
        at: 55800
        blackout_end: 5AM
        blackout_start: 2AM
        enabled: false
        frequency: 86400
        per_day: 5
      snapshot:
        all_for: 5
        at: 55800
        enabled: false
        frequency: 86400
        per_day: 5
    volumes:
    - name: test-vol
      size: 5G
  module_stderr: |-
    /Users/igorblackman/.local/share/virtualenvs/ansible-storage-7eUG76-U/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
      InsecureRequestWarning,
    /Users/igorblackman/.local/share/virtualenvs/ansible-storage-7eUG76-U/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
      InsecureRequestWarning,
    /Users/igorblackman/.local/share/virtualenvs/ansible-storage-7eUG76-U/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
      InsecureRequestWarning,
    /Users/igorblackman/.local/share/virtualenvs/ansible-storage-7eUG76-U/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
      InsecureRequestWarning,
    /Users/igorblackman/.local/share/virtualenvs/ansible-storage-7eUG76-U/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
      InsecureRequestWarning,
    /Users/igorblackman/.local/share/virtualenvs/ansible-storage-7eUG76-U/lib/python3.7/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
      InsecureRequestWarning,
    Traceback (most recent call last):
      File "<stdin>", line 102, in <module>
      File "<stdin>", line 94, in _ansiballz_main
      File "<stdin>", line 40, in invoke_module
      File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 205, in run_module
        return _run_module_code(code, init_globals, run_name, mod_spec)
      File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 96, in _run_module_code
        mod_name, mod_spec, pkg_name, script_name)
      File "/usr/local/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
        exec(code, run_globals)
      File "/var/folders/rn/m8sbwph1633czwjk22284shc0000gn/T/ansible_purestorage.flasharray.purefa_pgsched_payload_jqiqcpe4/ansible_purestorage.flasharray.purefa_pgsched_payload.zip/ansible_collections/purestorage/flasharray/plugins/modules/purefa_pgsched.py", line 435, in <module>
      File "/var/folders/rn/m8sbwph1633czwjk22284shc0000gn/T/ansible_purestorage.flasharray.purefa_pgsched_payload_jqiqcpe4/ansible_purestorage.flasharray.purefa_pgsched_payload.zip/ansible_collections/purestorage/flasharray/plugins/modules/purefa_pgsched.py", line 427, in main
      File "/var/folders/rn/m8sbwph1633czwjk22284shc0000gn/T/ansible_purestorage.flasharray.purefa_pgsched_payload_jqiqcpe4/ansible_purestorage.flasharray.purefa_pgsched_payload.zip/ansible_collections/purestorage/flasharray/plugins/modules/purefa_pgsched.py", line 221, in update_schedule
    TypeError: 'NoneType' object is not subscriptable
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1
```